### ISSUE TITLE
Fails if using pnpm

### ISSUE THREAD

**Issue by FFdhorkin** [cypress-io/github-action#145]
We are using pnpm (https://pnpm.js.org/) with our project. As such, we do not have a `package-lock.json` file. Even setting install to false fails with the cypress github action (see below).

Pnpm uses a `pnpm-lock.yaml` file that serves the same purpose, and generally (though not always) has the same syntax as npm. However, the lockfile formats are not compatible.

>This action installs local dependencies using lock files. If yarn.lock file is found, the install uses yarn --frozen-lockfile command. Otherwise it expects to find package-lock.json and install using npm ci command.

```
internal/fs/utils.js:220
    throw err;
    ^

Error: ENOENT: no such file or directory, open 'package-lock.json'
    at Object.openSync (fs.js:440:3)
    at Object.readFileSync (fs.js:342:35)
    at Function.module.exports.309.hasha.fromFileSync (/home/runner/work/_actions/cypress-io/github-action/v1.16.1/dist/index.js:4439:54)
    at Object.104 (/home/runner/work/_actions/cypress-io/github-action/v1.16.1/dist/index.js:2440:24)
    at __webpack_require__ (/home/runner/work/_actions/cypress-io/github-action/v1.16.1/dist/index.js:22:30)
    at startup (/home/runner/work/_actions/cypress-io/github-action/v1.16.1/dist/index.js:37:19)
    at /home/runner/work/_actions/cypress-io/github-action/v1.16.1/dist/index.js:41:18
    at Object.<anonymous> (/home/runner/work/_actions/cypress-io/github-action/v1.16.1/dist/index.js:44:10)
    at Module._compile (internal/modules/cjs/loader.js:959:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:995:10) {
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
  path: 'package-lock.json'
}
```

If relevant, there is a github action for pnpm: https://github.com/marketplace/actions/setup-pnpm
They also have these docs that might be relevant: https://pnpm.js.org/en/continuous-integration

**Comment by jennifer-shehane** [comment]
We would be open to PR to fix this. This would be pretty low priority for us to address ourselves since we have very few pnpm users.

**Comment by FFdhorkin** [comment]
I'll see if I can find some time to do that. I imagine it'll be a relatively simple PR.

**Comment by FFdhorkin** [comment]
Sorry for the delay. I've done most of the work for this PR, but I'm getting errors related to the `cache` dependency. I also see this locally - there's no `cache/lib/index`.

Is the `DEVELOPMENT.md` missing a step?

![image](https://user-images.githubusercontent.com/59899746/91360459-5897f900-e7ab-11ea-92b6-0c2803e535a2.png)


![image](https://user-images.githubusercontent.com/59899746/91360396-3d2cee00-e7ab-11ea-91c9-5d5cf13c3161.png)

If needed, you can peek at my WIP implementation of PNPM support for Cypress's github action here: https://github.com/FFdhorkin/github-action/tree/feature/pnpm-support

As an additional note, the actions unrelated to pnpm are failing at the same place. However, `example-recording / basic` fails and yet gets marked as success: https://github.com/FFdhorkin/github-action/runs/1033660717?check_suite_focus=true

**Comment by FFdhorkin** [comment]
@bahmutov any idea why I'm getting that import error?

**Comment by bahmutov** [comment]
Could you rebase from the latest code - we have started using the official `actions/cache` NPM dependency, so this error should be gone.

**Comment by tnrich** [comment]
@FFdhorkin seems close! Would be much appreciated if you could fix this :)

**Comment by penx** [comment]
I've rebased off master and opened a draft PR https://github.com/cypress-io/github-action/pull/381

**Comment by penx** [comment]
I've also opened a draft PR #381 to solve a similar issue using npm workspaces #377 that allows the lock file path to be specified, but it currently errors with this:

```
The `npm ci` command can only install with an existing package-lock.json
```

**Comment by fc1943s** [comment]
Just installed pnpm and got blocked by this ;(
I was using yarn before, and it was working, even though the image specifies that the github action is expecting npm's package-lock.json file. 

![image](https://user-images.githubusercontent.com/688618/126344244-9b1d3f48-2dde-4aeb-b0ce-74aef3a33df2.png)

@penx is there a way of using your draft branch for now, changing `- uses: cypress-io/github-action@v2` for `- uses: penx/github-action@4c488bcaa610e82cc1b45c3b0ac935e415fc6e59` or something like that?

**Comment by tcrwt** [comment]
I got stuck on this but running the pnpm install and cypress separately seemed to resolve it. It's possible this isn't the best way but it got things working!

```yaml
- name: Set up pnpm
  uses: pnpm/action-setup@v2.0.1
  with:
    version: 6
    # Only install what we need to run the client in-action and tests against it
    # format:    `--filter <packagename>...`
    run_install: |
      - args: [--filter myclient... --filter cypress-tests...]

- name: Run Cypress tests
  uses: cypress-io/github-action@v2
  with:
    # we have already installed all dependencies above
    install: false
    command: pnpm --dir cypress/ cypress:ci
    start: pnpm --dir myclient/ dev
    wait-on: 'https://localhost:3000'
```

**Comment by DavidWells** [comment]
I also just ran into this. We are using `pnpm` for faster installs.

Whats odd is `/home/runner/work/clients/clients/src/app/package-lock.json` seems like the wrong path (notice the double `client` in that path). It seems like it should be looking for `/home/runner/work/clients/src/app/package-lock.json` (correct path single `clients`). But yeah no `package-lock.json` would exist there either.

```
internal/fs/utils.js:220
    throw err;
    ^

Error: ENOENT: no such file or directory, open '/home/runner/work/clients/clients/src/app/package-lock.json'
    at Object.openSync (fs.js:440:3)
    at Object.readFileSync (fs.js:342:35)
    at Function.module.exports.hasha.fromFileSync (/home/runner/work/_actions/cypress-io/github-action/v2/dist/index.js:18048:54)
    at lockHash (/home/runner/work/_actions/cypress-io/github-action/v2/dist/index.js:6729:26)
    at getNpmCache (/home/runner/work/_actions/cypress-io/github-action/v2/dist/index.js:6739:16)
    at restoreCachedNpm (/home/runner/work/_actions/cypress-io/github-action/v2/dist/index.js:6784:21)
    at installMaybe (/home/runner/work/_actions/cypress-io/github-action/v2/dist/index.js:7365:5)
    at Object.<anonymous> (/home/runner/work/_actions/cypress-io/github-action/v2/dist/index.js:7388:1)
    at __webpack_require__ (/home/runner/work/_actions/cypress-io/github-action/v2/dist/index.js:24:31)
    at startup (/home/runner/work/_actions/cypress-io/github-action/v2/dist/index.js:43:19) {
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
  path: '/home/runner/work/clients/clients/src/app/package-lock.json'
}
```

Setting `install: false` to false fixes this for us

**Comment by Kocal** [comment]
A PR has been opened #586
